src/test/java/com/epam/wca/gym/service/ProfileServiceImplTest.java

```diff
@@ -0,0 +1,65 @@
+package com.epam.wca.gym.service;
+
+import com.epam.wca.gym.service.impl.ProfileServiceImpl;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class ProfileServiceImplTest {
+    @Mock
+    private Map<String, Integer> usernameCounter;
+
+    @InjectMocks
+    private ProfileServiceImpl profileService;
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+    }
+
+    @Test
+    void testCreateUserName_NewUser() {
+        String firstName = "John";
+        String lastName = "Doe";
+        String expectedUsername = "John.Doe";
```

Comment:
Use parameterized tests where possible to test multiple scenarios in a single test case.
```suggestion
@ParameterizedTest
@CsvSource({ "John, Doe, John.Doe", "Jane, Smith, Jane.Smith" })
void testCreateUserName(String firstName, String lastName, String expectedUsername) {
    assertEquals(expectedUsername, profileService.createUserName(firstName, lastName));
}
```

Reply:
Done